tensorflow/compiler/jit/deadness_analysis.cc

```diff
@@ -1069,6 +1240,52 @@ Status DeadnessAnalysisImpl::PopulateWithReversePostOrder(
         absl::StrJoin(unreachable_nodes, ", "));
   }
 
+  std::vector<Node*> tpo;
+  TF_RETURN_IF_ERROR(
+      GetFrameBasedTopologicalOrder(&graph_, control_flow_info_, &tpo));
+
+  size_t frame_start = 0;
+  for (size_t i = 0; i < tpo.size(); ++i) {
+    // Collect nodes until we see a node who has a different root frame.
+    if (i != tpo.size() - 1) {
+      string i_frame_name, next_frame_name;
+      TF_RETURN_IF_ERROR(GetRootFrame(tpo[i], control_flow_info_,
+                                      graph_.source_node(), &i_frame_name));
+      TF_RETURN_IF_ERROR(GetRootFrame(tpo[i + 1], control_flow_info_,
+                                      graph_.source_node(), &next_frame_name));
+      if (i_frame_name == next_frame_name) {
```

Comment:
This is a bit oddly written, how about restructuring the loop as:

```
frame_start = ...;
while (frame_start != tpo.size()) {
  for (i = frame_start + 1; ...) {
    // Find the first i such that frame_of[tpo[i]] != frame_of[tpo[frame_start]]
  }
  sub_tpo = ...
  frame_start = i + 1;
}
```

Reply:
Will do.

Reply:
Done.